bpy_speckle/converter/to_speckle/to_speckle.py

```diff
@@ -25,9 +26,9 @@ def convert_to_speckle(
             blender_object, blender_object.data, scale_factor, units
         )
         if meshes:
-            for mesh in meshes:
-                if hasattr(mesh, "applicationId"):
-                    mesh.applicationId = None
+            # Assign unique applicationIds to each submesh using the centralized system
+            for i, mesh in enumerate(meshes):
+                mesh.applicationId = set_submesh_id(blender_object, i)
```

Comment:
With this code right now, it will have the id.

`OBJECT:{objectname}_mat{i}` right?
do we not want them to have
`MESH:{meshname}_mat{i}` right?

i.e. `set_submesh_id(mesh, i)`

(and I'd probably suggest putting this inside the `mesh_to_speckle_meshes` function)